### PR TITLE
Add show methods for DiffDelta, DiffFile and test

### PR DESCRIPTION
**Pull request by kshyatt**
Made the consts an enum so the printing would Just Work.

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -516,6 +516,11 @@ mktempdir() do dir
                 @test LibGit2.isdiff(repo, "HEAD")
                 @test LibGit2.isdirty(repo, cached=true)
                 @test LibGit2.isdiff(repo, "HEAD", cached=true)
+                diff = LibGit2.diff_tree(repo, LibGit2.GitTree(repo, "HEAD^{tree}"), "", cached=true)
+                diff_strs = split(sprint(show, diff[1]), '\n')
+                @test diff_strs[1] == "DiffDelta:"
+                @test diff_strs[2] == "Status: DELTA_MODIFIED"
+                @test diff_strs[3] == "Number of files: 2"
```
Comment (ararslan) [inline test/libgit2.jl]: Can you add a test for `DiffFile` as well?

Comment (kshyatt) [inline test/libgit2.jl]: Well, it's implicitly tested here, but yes I can add some checks for the `DiffFile` part of this output.

Comment (ararslan) [inline test/libgit2.jl]: If it's already being implicitly tested, I'll leave that up to you as to whether you think further testing is necessary. 🙂

Comment (kshyatt) [inline test/libgit2.jl]: MORE TESTS ALWAYS MORE TESTS WHEN YOUR NUMBER OF TESTS GOES UP YOU'RE HAVING MORE FUN

**Comment by kshyatt** [comment]
@ararslan ok with you?

**Review by ararslan — APPROVED** [review]
Just needs a conflict resolution

**Comment by kshyatt** [comment]
Yep, resolving now.

**Comment by kshyatt** [comment]
CI passed, going to press the pretty button.

**Pull request merged**